src/Common/OpenTelemetryTraceContext.cpp

```
@@ -294,14 +294,29 @@ TracingContextHolder::~TracingContextHolder()
 {
     if (this->root_span.isTraceEnabled())
     {
-        auto shared_span_log = current_thread_trace_context.span_log.lock();
-        if (shared_span_log)
+        try
         {
-            this->root_span.addAttribute("clickhouse.thread_id", getThreadId());
-            this->root_span.finish_time_us
-                = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
-
-            shared_span_log->add(OpenTelemetrySpanLogElement(this->root_span));
+            auto shared_span_log = current_thread_trace_context.span_log.lock();
+            if (shared_span_log)
+            {
+                try
+                {
+                    this->root_span.addAttribute("clickhouse.thread_id", getThreadId());
+                }
+                catch (...)
```

Is a nested try/catch (l. 297 and l. 302) really necessary or can't we just do with the outer try/catch. It is not really clear to the reader why exceptions are ignored in the inner catch but not in the outer catch.

---

The implicit memory allocation in `addAttribute` might throws exception, I think this exception could be ignored because it's acceptable that such attribute is not logged in the final result. 

If we don't keep the nested try-catch here, the whole log is missing.

---

Now the CI reports that `'Memory tracker: allocations not allowed.'.`

Seems that ClickHouse prohibits memory allocation in destructor in Debug Release. Can't remeber why we put this code block in the dtor, but it's OK to be placed in ctor.

---

There is no general rule that dtors may not allocate in Debug builds. The error is caused by `DENY_ALLOCATIONS_IN_SCOPE` at the beginning of `ThreadPoolImpl<Thread>::worker()`.

---

Sorry for the late update. 

In the lastest commit, we also fixed a problem that tracing context might be initialized twice for `ThreadPool<ThreadFromGlobalPool>`.  A dedicated `Thread4ThreadPool`( in ThreadPool.h ), which inherits  `ThreadFromGlobalPool` is created, to solve this problem. 

Please let me know if there's any problem.